addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -112,6 +127,17 @@ def _compute_display_name(self):
                 channel_name=member.channel_id.display_name,
             )
 
+    @api.depends("last_interest_dt", "unpin_dt", "channel_id.last_interest_dt")
+    def _compute_is_pinned(self):
+        for member in self:
+            member.is_pinned = not member.unpin_dt or \
+                (member.last_interest_dt and member.last_interest_dt > member.unpin_dt) or \
+                (member.channel_id.last_interest_dt and member.channel_id.last_interest_dt > member.unpin_dt)
+            
+    def _inverse_is_pinned(self):
+        for member in self:
+            member.unpin_dt = False if member.is_pinned else fields.Datetime.now()
```

Comment:
Shouldn't this write `fields.Datetime.now()` only if `unpin_dt` doesn't already have a value?

Reply:
at some point, we will unpin it again so we need to update the date we unpin the channel ?